Accurate temperature alerts

Errant-4 opened the pull request:
## About the PR
Temperature alerts are now counted from the actual dangerous cold and dangerously hot temperatures for each character depending on their species, instead of the same human range being used in every case

## Why / Balance
The current temperature alert range is hardcoded. This means that for reptilians and moths the critical temperature warnings do not line up with the actual temperatures where they start taking damage.

The new warning ranges also reveal the previously not apparent fact that reptilian and moth body temperatures are actually within their Severity 1 temperature warnings (if we were to apply the same warning ranges to them). Even if they wear insulation, their body temperature will "stabilise" in a spot where they get a temperature warning. As a solution, the alert ranges are now scaled between the ideal temperature and the temperature thresholds

## Technical details
(The previous final writeup candidate of this was more thorough, I swear, but then a crash ate the entire thing while I was still editing it, so this is what you get now)

Warnings for arachnids, diona, dwarves, humans and slimes are effectively unchanged. The new ranges for moths and reptilians are:

![Screenshot from 2023-09-10 12-48-56](https://github.com/space-wizards/space-station-14/assets/35878406/a67bd2e3-2dc0-4994-8503-3df0d96b40ac)

Mobs that don't have a normal temperature due to lacking ThermalRegulatorComponent, or have theirs incorrectly set beyond their temp threshold are calculated a new ideal temperature whether they like it or not, averaged from their temp thresholds. They will possibly constantly get temperature warnings anyway but they should have thought about that before materialising.
This does not affect any existing mobs as far as I could tell. I thought Maintenance Drones would get constant alerts, since they are given a TemperatureComponent unlike any other Silicon (with a ludicrous 5000 kelvin upper threshold). But maintenance drones are disabled and probably need a prototype review anyway

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Errant
- fix: Temperature alerts are now more accurate for different species.

LankLTE commented:
I’m a fan of this change, but Moths and Reptiles would need to have their solution first (ideally making their temperature warning ranges lower)

Partmedia commented on Content.Server/Temperature/Systems/TemperatureSystem.cs:
```diff
@@ -139,42 +139,66 @@ private void OnRejuvenate(EntityUid uid, TemperatureComponent comp, RejuvenateEv
-            case <= 260:
-                _alertsSystem.ShowAlert(uid, AlertType.Cold, 3);
-                break;
+            var cold3 = temperature.ColdDamageThreshold;
```
Instead of hard coding the difference between each step, then checking later to see if the ranges overlap, why isn't this just divide the difference between HeatDamageThreshold and ColdDamageThreshold into several ranges?

Errant-4 replied:
I wanted to keep the width of the individual warning bands absolute, not relative. That seemed more "consistent" to me

Errant-4 replied:
I came around to see that dynamic alert temperatures are better so I vultured your code, thank you

Errant-4 commented:
I have added a mockup of the suggested squashed warning bands to the image

LankLTE commented:
I think the squashed warning bands look fine.

Errant-4 commented:
Since dynamically calculated and squashed ranges are not actually mutually exclusive with possibly changing the normal temperatures, there is no need for me to hold up this PR, I can go and die on that hill in that other PR
Alert temperature points are now actually calculated for each range, I have updated the opening notes

deltanedas commented:
:trollface:

Partmedia commented:
Sorry for taking so long to get back to this. Could I ask you to please remove the "namespace fix" and "minor code tweaks and fixes" from this PR?

Partmedia commented:
@deltanedas #20662 will conflict with this, so please help the submitter fix merge conflicts. Or, I can mark the temperature refactor as Do Not Merge until this change is in.

deltanedas commented:
> so please help the submitter fix merge conflicts. 

i cant make changes on other peoples prs im not a maint

Errant-4 commented:
> Sorry for taking so long to get back to this. Could I ask you to please remove the "namespace fix" and "minor code tweaks and fixes" from this PR?

Was that only for managing the merge conflict? Is this good now?

deltanedas commented:
refactor already merged

Errant-4 commented:
> refactor already merged

I just wanted to double-check that the merge conflict with your refactor was the only reason for that request, so it's no longer necessary to remove any of the mentioned parts

Partmedia commented on Content.Server/Temperature/Systems/TemperatureSystem.cs:
```diff
@@ -27,11 +28,11 @@ public sealed class TemperatureSystem : EntitySystem
     public float UpdateInterval = 1.0f;
 
-    private float _accumulatedFrametime;
+    private float _accumulatedFrameTime;
```
There are still a lot of other unrelated changes like this to the content of your PR. We can still review with changes like this, but it makes it more difficult and as a result PRs can get buried like this one.

The PR guidelines are designed to make PRs easy to review so that actual content changes are clearly separate from no-op changes like this and so that PRs get in faster. Your help with this is appreciated.

Errant-4 replied:
Ok I see, thank you for clarifying. I did not actually plan any of those changes going in, the thought process was that if I am touching this system I should just as well clean up every thing Rider was warning and nagging about, so the next person gets to work with a pristine file with a more easily readable overview. 
I figured that putting them in a distinct commit before the important changes will be enough, but I'll move them to their own PR then

deltanedas commented on Content.Server/Temperature/Systems/TemperatureSystem.cs:
```diff
@@ -27,11 +28,11 @@ public sealed class TemperatureSystem : EntitySystem
 
     public float UpdateInterval = 1.0f;
 
-    private float _accumulatedFrametime;
```
presumably this too

deltanedas replied:
pro sniped

Errant-4 replied:
Thank you

Partmedia reviewed (approved):
The core of these changes look good. Because of the other unrelated changes, I'm going to loop back to this to test this in-game.

Partmedia commented:
Thank you for your contribution.

Errant-4 commented:
> Thank you for your contribution.

If my changelog text is not great it would probably be helpful to at least tell me about it so I'm aware I need to make it more concise next time, instead of leaving it to fate if I spot the change or not
